sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -88,16 +77,26 @@ public void deleteTable(String tableName) {
         return client.deleteTableWithResponse(tableName, context).block();
     }
 
+    /**
+     * query all the tables under the storage account
+     *
+     * @return a list of tables that meet the query
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedIterable<TableItem> listTables() {
+        return new PagedIterable<>(client.listTables(new ListTablesOptions()));
```

Comment:
I'd consider making `new ListTablesOptions()` a member variable so we don't have to allocate a new object to it for every call.